### ISSUE TITLE
Loss goes Nan after 1st update

### ISSUE THREAD

**Issue by vfdev-5** [NVIDIA/apex#651]
Hi 

I'm retrying to train CycleGAN implementation with AMP (first train was successful in May 2019 and apex of that time).

According to opt level I have various errors:
"O1" : RuntimeError: CUDA error: an illegal memory access was encountered
"O2" : loss of generators goes NaN
and
"O0" : all losses are correct and training works

I also tried to split and scale separately the losses of generators, but it does not help.

Here is the script to reproduce the error with 3 updates in the end of the file:
https://gist.github.com/vfdev-5/7a651ca550843737200f148755c42861

Basically it is just python script of this notebook: https://github.com/pytorch/ignite/blob/master/examples/notebooks/CycleGAN.ipynb

Any suggestions, please

cc @ptrblck

**Comment by vfdev-5** [comment]
Fixed by https://github.com/pytorch/pytorch/pull/37569
Here is a training colab:
https://colab.research.google.com/drive/1P2D_bYRn9l3I6t1veaKOsPWyxU-Y9-Jg?usp=sharing
